admin/services/school-impersonation.service.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+const R = require('ramda')
+
+const schoolDataService = require('../services/data-access/school.data.service')
+const schoolImpersonationDfeNumberValidator = require('../lib/validator/school-impersonation-dfe-number-validator')
+
+const schoolImpersonationService = {}
+
+/**
+ * Validate and assign school to helpdesk session
+ * @param {object} user
+ * @param {string} dfeNumber
+ * @returns {Object}
+ */
+schoolImpersonationService.validateImpersonationForm = async (user, dfeNumber) => {
+  let validationError
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberEmpty(dfeNumber)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  let school
+  try {
+    school = await schoolDataService.sqlFindOneByDfeNumber(dfeNumber)
+  } catch (error) {}
+  validationError = schoolImpersonationDfeNumberValidator.isDfeNumberValid(school)
+  if (validationError.hasError()) {
+    return validationError
+  }
+  user.School = school.dfeNumber
+  user.schoolId = school.id
+  user.timezone = school.timezone
+  return user
+}
+
+/**
+ * Remove impersonation data from request.user object
+ * @param {object} user
+ * @returns {void}
+ */
+schoolImpersonationService.removeImpersonation = (user) => {
+  R.forEach(k => delete user[k], ['School', 'schoolId', 'timezone'])
```

Comment:
Should we add a message to the user?

Reply:
@GeorgeChatzigiannis - do you think it is worth adding a flash message for the user?

Reply:
@jon-shipley it's already included. Please have a look at the helpdesk controller method `postRemoveSchoolImpersonation`.